src/lib/operators/table_scan.cpp

```diff
@@ -203,23 +218,28 @@ std::shared_ptr<const Table> TableScan::_on_execute() {
   return std::make_shared<Table>(in_table->column_definitions(), TableType::References, std::move(output_chunks));
 }
 
-std::shared_ptr<AbstractExpression> TableScan::_resolve_uncorrelated_subqueries(
-    const std::shared_ptr<AbstractExpression>& predicate) {
+std::shared_ptr<const AbstractExpression> TableScan::_resolve_uncorrelated_subqueries(
+    const std::shared_ptr<const AbstractExpression>& predicate) {
   // If the predicate has an uncorrelated subquery as an argument, we resolve that subquery first. That way, we can
   // use, e.g., a regular ColumnVsValueTableScanImpl instead of the ExpressionEvaluator. That is faster. We do not care
   // about subqueries that are deeper within the expression tree, because we would need the ExpressionEvaluator for
   // those complex queries anyway.
 
-  if (!std::dynamic_pointer_cast<BinaryPredicateExpression>(predicate) &&
-      !std::dynamic_pointer_cast<IsNullExpression>(predicate) &&
-      !std::dynamic_pointer_cast<BetweenExpression>(predicate)) {
+  if (!std::dynamic_pointer_cast<const BinaryPredicateExpression>(predicate) &&
+      !std::dynamic_pointer_cast<const IsNullExpression>(predicate) &&
+      !std::dynamic_pointer_cast<const BetweenExpression>(predicate)) {
     // We have no dedicated Impl for these, so we leave them untouched
     return predicate;
   }
 
-  auto new_predicate = predicate->deep_copy();
-  for (auto& argument : new_predicate->arguments) {
-    const auto subquery = std::dynamic_pointer_cast<PQPSubqueryExpression>(argument);
+  // ToDo(Julian) Discuss
+  // We do not want to deep copy any PQPs, because they become irrelevant after this function. Instead, we could
+  //  (1) materialize the uncorrelated subqueries and
+  //  (2) manually create a new predicate from it.
```

Comment:
Sounds good. What would happen if the same PQP is used in two places?

Reply:
I changed it successfully.

> Sounds good. What would happen if the same PQP is used in two places?

Nothing I guess. The TableScan simply deregisters itself from the PQP.